src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticSourceEventSource.cs

```diff
@@ -641,6 +641,8 @@ public FilterAndTransform(string filterAndPayloadSpec, int startIdx, int endIdx,
                         [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2119",
                             Justification = "DAM on EventSource references this compiler-generated local function which calls a " +
                                             "method that requires unreferenced code. EventSource will not access this local function.")]
+                        [UnconditionalSuppressMessage("AotAnalysis", "IL3050:RequiresDynamicCode",
+                            Justification = "Activity's properties are being preserved with the DynamicDependencies on OnActivityStarted.")]
```

Comment:
I assume that this warning is due to MakeGenericType below. I do not understand how this message justifies that it is safe for AOT.

Reply:
Sorry, this is a copy and paste error, it should be the same (modified for IL3050) as the IL2026 suppress message (that `DiagnosticSource.Write` is marked with RUC)